infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -222,6 +228,12 @@ public void sendWithBuffering(List<LogReplicationEntryMsg> dataToSend, String me
     public LogReplicationEntryMsg resend() {
         LogReplicationEntryMsg ack = null;
         boolean force = false;
+
+        // Wait before retrying
+        if (isBackpressureActive.get()) {
```

Comment:
I think this check and wait must be done before sending, after processing the acks on L238